specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,383 @@
+                        InsertMenu(hPopupMenu, 0, MF_BYPOSITION | MF_STRING, id, name);
+                        BOOL hasNext = FALSE;
+                        CHECK_FAILURE(iterator->MoveNext(&hasNext));
+                    }
+                    HWND hWnd;
+                    m_appWindow->GetWebViewController()->get_ParentWindow(&hWnd);
+                    SetForegroundWindow(hWnd);
+                    POINT p;
+                    CHECK_FAILURE(args->get_Location(&p));
+                    UINT selectedItem = TrackPopupMenu(hPopupMenu, TPM_TOPALIGN | TPM_LEFTALIGN | TPM_RETURNCMD, p.x, p.y, 0, hWnd, NULL);
+                    CHECK_FAILURE(args->put_SelectedItem(selectedItem));
+                }
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+                m_sampleWindow->RunAsync([deferral, showMenu]() {
+                    showMenu();
+                    CHECK_FAILURE(deferral->Complete());
+                });
+                return S_OK;
+            }).Get(),
+        &m_contextMenuRequestedToken);
+```
+## .Net/ WinRT Add/ Remove Entries From Browser Menu
+
+## .Net/ WinRT Use Data to Display Custom Context Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        CoreWebView2Deferral deferral = args.GetDeferral();
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+            using (deferral)
+            {
+                CoreWebView2ContextMenuItemCollection collection = args.MenuItems;
+                CoreWebView2SelectionContext context = args.context;
+                CoreWebView2ContextMenuItemCollectionIterator iterator = collection.GetIterator();
+                args.Handled = true;
+                ContextMenu cm = this.FindResource("ContextMenu") as ContextMenu;
+                cm.Items.Clear();
+                while (iterator.HasCurrent)
+                {
+                    CoreWebView2ContextMenuItem current = iterator.GetCurrent();
+                    MenuItem newItem = new MenuItem();
+                    newItem.Header = current.Name;
+                    newItem.Click += (s, ex) => args.SelectedItem = current.id;
+                    cm.Items.Add(newItem);
+                    iterator.MoveNext();
+                }
+                cm.IsOpen = true;
+            }
+        }, null);
+    };
+```
+# Remarks
+
+# API Notes
+
+# API Details
+ ```cpp
+    interface ICoreWebView2_4;
+    interface ICoreWebView2ContextMenuItem;
+    interface ICoreWebView2ContextMenuItemCollection;
+    interface ICoreWebView2ContextMenuItemCollectionIterator;
+    interface ICoreWebView2ContextMenuRequestedEventArgs;
+    interface ICoreWebView2ContextMenuRequestedEventHandler;
+
+    /// Defines the Context Menu Items
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_ID {
```

Comment:
Can we give this a more specific name than just Id? Usually we think of Ids as uniquely identifying particular menu items but that's not the case here. These are more like ids for built in menu items. Or the menu item kind/type rather than an id.